internal/resources/network_peer.go

```diff
@@ -398,6 +459,13 @@ func (n *NetworkPeer) getNetworkPeer(ctx context.Context, organizationId, projec
 		return nil, fmt.Errorf("%s: %w", errors.ErrUnmarshallingResponse, err)
 	}
 
+	azure, err := networkResp.AsAZURE()
+	if err == nil && azure.AzureConfigData.AzureTenantId != "" {
+		networkResp.ProviderType = "azure"
+	} else if err != nil {
+		return nil, fmt.Errorf("%s: %w", errors.ErrReadingAzureConfig, err)
+	}
+
```

Comment:
If err == nil and zaure.AzureConfigData.AzureTenantId is an empty string, should we return an error in this case?, seems like this case is missing.

We can also use a switch case statement instead of an if-else statement in this case.

Reply:
@aniket-Kumar-c thanks for the review. Here we add this condition to just set the networkResp.ProviderType = "azure" explicitly in TF as we don't receive it from v4 get response, hence we check if the AzureTenantId is populated then set the provider type as azure. What's the point of checking if AzureTenantId is empty? In that case, we need to check the other fields in the providerConfig also - `Cidr`, `ResourceGroup`, `SubscriptionId`, `VnetId` are empty too. These would never be empty if the record is present. So is this needed?